fix(docs): re-ordered small paragraphs, clarified wording, and added links to tech homepages

JoeHCQ1 opened the pull request:
# Summary

As a new Unicorn, I made changes which I believe would've increased my comprehension of this document on the first read-through.

## Description

- Re-ordered tools in monitoring to build conceptually in line with the way each enables the next
- Turned first mention of each tool into a link to the relevant docs. Both an ease of reference change and also to disambiguate in the event a name collision exists in the reader's mind.
- Clarified relationship between AuthService and Keycloak as this Unicorn initially saw a redundancy and suspected AuthService was perhaps the service IAM solution while Keycloak was the user IAM solution. We could link out to BigBang's explanation of this relationship but I did not: https://docs-bigbang.dso.mil/2.2.0/docs/understanding-bigbang/package-architecture/authservice/
- Wording changes for precision. Example, in MetricsServer, speaking of "container" metrics is more precise (and I believe more accurate) than "cluster" metrics. If it reports on nodes and containers I would (personally) speak of both over the more nebulous "cluster" (which has many more primitives (notably servics and pods) which could be but are not (I think) being reported on).
- Most likely errors needing correction:
  - Metrics Server may be misunderstood
  - AuthService may be better characterized but I didn't want to plagarize the BigBang docs, and am new to this service.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

JoeHCQ1 committed:
Comprehension-easing improvements:
- Re-ordered tools in monitoring to build conceptually in line with the way each enables the next
- Turned first mention of each tool into a link to the relevant docs. Both an ease of reference change and also to disambiguate in the event a name collision exists in the reader's mind.
- Clarified relationship between AuthService and Keycloak as this Unicorn initially saw a redundancy and suspected AuthService was perhaps the service IAM solution while Keycloak was the user IAM solution. We could link out to BigBang's explanation of this relationship but I did not: https://docs-bigbang.dso.mil/2.2.0/docs/understanding-bigbang/package-architecture/authservice/
- Wording changes for precision. Example, in MetricsServer, speaking of "container" metrics is more precise (and I believe still accurate) than "cluster" metrics. If it reports on nodes and containers I would (personally) speak of both over the more nebulous "cluster" (which has many more primitives (notably servics and pods) which could be but are not (I think) being reported on).
- Most likely errors needing correction:
  - Metrics Server may be misunderstood
  - AuthService may be better characterized but I didn't want to plagarize the BigBang docs, and am new to this service.

JoeHCQ1 committed:
Merge branch 'main' into fix/re-order-clarify-and-add-links-to-application-baseline

mjnagel commented on docs/application-baseline.md:
```diff
@@ -16,10 +16,10 @@ For optimal deployment and operational efficiency, it is important to deliver a
+| **Security and Compliance**        | **[NeuVector](https://open-docs.neuvector.com/):** Offers container-native security, protecting applications against threats and vulnerabilities.<br><br> **[Pepr](https://pepr.dev/):** UDS policy engine and operator for enhanced security and compliance.|
+| **Identity and Access Management** | **[Keycloak](https://www.keycloak.org/):** A robust open-source Identity and Access Management solution, providing centralized authentication, authorization, and user management for enhanced security and control over access to mission-critical resources.|
+| **Backup and Restore**             | **[Velero](https://velero.io/):** Provides backup and restore capabilities for Kubernetes clusters, ensuring data protection and disaster recovery.|
+| **Authorization**                  | **[AuthService](https://github.com/istio-ecosystem/authservice):** Offers centralized authorization services, managing access control and permissions within the Istio mesh. AuthService plays a supporting role to Keycloak as it handles part of the OIDC redirect flow.
```
```suggestion
| **Authorization**                  | **[AuthService](https://github.com/istio-ecosystem/authservice):** Offers centralized authorization services, managing access control and permissions within the Istio mesh. AuthService plays a supporting role to Keycloak as it handles part of the OIDC redirect flow.|
```

mjnagel commented on docs/application-baseline.md:
```diff
@@ -16,10 +16,10 @@ For optimal deployment and operational efficiency, it is important to deliver a
-| **Backup and Restore**             | **Velero:** Provides backup and restore capabilities for Kubernetes clusters, ensuring data protection and disaster recovery.                                                                                                                                                                                                                         |
-| **Authorization**                  | **AuthService:** Offers centralized authorization services, managing access control and permissions within the mission environment.                                                                                                                                                                                                                   |
+| **Service Mesh**                   | **[Istio](https://istio.io/):** A powerful service mesh that provides traffic management, load balancing, security, and observability features.                                                                                                                                                                                                                       |
+| **Monitoring**                     | **[Metrics Server](https://kubernetes-sigs.github.io/metrics-server/):** Provides container resource utilization metrics API for Kubernetes clusters.<br><br>**[Prometheus](https://prometheus.io/):** Scrapes Metrics Server API and stores the data in a time-series database for insights into container health and performance.<br><br> **[Grafana](https://grafana. ...
```
I think the prometheus description should also capture application metrics (not just metrics server api) since we use it to grab more than just the metrics-server? The before state might not be perfect but does capture that part for prometheus a bit, maybe can steal some wording from there.

JoeHCQ1 replied:
@mjnagel want to re-check my wording? I added a reference to application metrics, and the goal of these data points from "container health" back to "application health" as it had been.

mjnagel replied:
LGTM!

mjnagel reviewed (commented):
Appreciate the contribution - few small comments! @UnicornChance may be good to validate for me that nothing changes here with the expected format on the docs site side.

JoeHCQ1 committed:
fix: add clause that prometheus also gathers application metrics (not just metrics server API)

Addresses: https://github.com/defenseunicorns/uds-core/pull/531/files/67091de19b7dbaa9edd35f5ad4a738c2fa682935#r1663103470

JoeHCQ1 committed:
fix: added suggested pipe to complete markdown table

Co-authored-by: Micah Nagel <micah.nagel@defenseunicorns.com>

mjnagel committed:
Merge branch 'main' into fix/re-order-clarify-and-add-links-to-application-baseline
